lib/exec.js

```diff
@@ -66,3 +68,82 @@ module.exports = function ({
 
     return p
 }
+
+function spawn (cmd, args, opts) {
+    reporter.debug(cmd, args, opts)
+    return xspawn.sync(cmd, args, {
+        stdio: 'inherit',
+        ...opts,
+    })
+}
+
+function run (cmd, { args, opts }, callback) {
+    return handleRun(
+        spawn(cmd, args, {
+            stdio: 'inherit',
+            ...opts,
+        }),
+        callback
+    )
+}
+
+function bin (cmd, { args, cwd = process.cwd(), opts }, callback) {
+    const nodemodules = findup.sync('node_modules', {
+        cwd,
+        type: 'directory',
+        allowSymlinks: true,
+    })
+
+    const binCmd = path.join(nodemodules, '.bin', cmd)
+
+    return handleRun(
+        spawn(binCmd, args, {
+            stdio: 'inherit',
+            ...opts,
+        }),
+        callback
+    )
+}
+
+function callback () {
+    let status = 0
+    return result => {
+        if (!result) return status
+
+        if (result.status > status) {
+            status = result.status
+        }
+    }
+}
+
+function exit (code, msg) {
+    if (msg && code > 0) {
+        reporter.print('')
+        reporter.error(msg)
+    }
+    process.exit(code)
+}
```

Comment:
I like having this function!  Should it maybe live somewhere else though?  Seems a bit unrelated to the execution of child processes?

Reply:
Yeah, it is a bit unrelated. We could have a "process" module that wraps process stuff. Another case would be to wrap the `process.cwd()` command to allow setting a custom `cwd` instead of rolling a custom solution for that in each tool.

Reply:
8d973fc